src/EntityFramework.Storage/Configuration/ServiceCollectionExtensions.cs

```diff
@@ -36,24 +36,40 @@ public static class IdentityServerEntityFrameworkBuilderExtensions
         /// <param name="storeOptionsAction">The store options action.</param>
         /// <returns></returns>
         public static IServiceCollection AddConfigurationDbContext<TContext>(this IServiceCollection services,
-        Action<ConfigurationStoreOptions> storeOptionsAction = null)
-        where TContext : DbContext, IConfigurationDbContext
+            Action<ConfigurationStoreOptions> storeOptionsAction = null)
+            where TContext : DbContext, IConfigurationDbContext
         {
             var options = new ConfigurationStoreOptions();
             services.AddSingleton(options);
             storeOptionsAction?.Invoke(options);
 
             if (options.ResolveDbContextOptions != null)
             {
-                services.AddDbContext<TContext>(options.ResolveDbContextOptions);
+                if (options.EnablePooling)
+                {
+                    services.AddDbContextPool<TContext>(options.ResolveDbContextOptions, options.PoolSize);
+                }
+                else
+                {
+                    services.AddDbContext<TContext>(options.ResolveDbContextOptions);
+                }
             }
             else
             {
-                services.AddDbContext<TContext>(dbCtxBuilder =>
+                if (options.EnablePooling)
+                {
+                    services.AddDbContext<TContext>(dbCtxBuilder =>
```

Comment:
I'm new to ef connection pooling, but isn't this `if` logic mixed up with below?

Reply:
Good catch ty